doc/architecture/configuration-and-management.rst

```diff
@@ -0,0 +1,203 @@
+-------------------
+
+The *Ostro* configuration process is done in two major phase:
+
+-  synchronisation & delivery
+
+-  installation
+
+Agents that do synchronization & delivery related tasks are called
+*Forwarders* while the agent that take the actual configuration into
+effect is the *Installer*.
+
+The interface between *Forwarders* and the *Installer* called *Drop
+Zone*. *Drop Zone* is implemented as directory tree, eg. under
+*/var/cache/confs,* and it hosts all the configuration data in various
+configuration files called *Configuration Fragment*\ s .
+
+The *Configuration Fragment*\ s are either *JSON* or *XML* strings.
+Future releases might also support *.ini* formatted strings.
+
+|image0|
+
+*Forwarders*
+~~~~~~~~~~~~
+
+*Forwarders*, that are responsible to keep up-to-date the configuration
+files in the *Drop Zone*, usually syncing/fetching some cloud maintained
+data. Multiple *Forwarders* can be used simultaneously to maintain
+configuration data subsets in the *Drop Zone*. An actual file, however,
+should always be maintained by a single *Forwarder*.
+
+The *Drop Zone,*\ that is a directory tree conatining *JSON* or *XML*
+formatted text files, can be trivially mapped to hierarchical
+*Key/Value* pairs where the *Key* is the path to the *Configuration
+Fragement* and the *Value* is the fragment's *JSON* or *XML*
+representation. Such way, beside the various file distribution
+mechanisms and cloud based file sharing services, *Key/Value* tree based
+distribution & synchronization systems (like *etcd*) can also be used as
+*Forwarders*.
+
+The simplest *Forwarder* is the *cp* or *scp* utility that can be used
+to deliver a *Configuration Fragment* into the *Drop Zone. NFC*\ agents
+or *etcd* are examples for more complex *Forwarders*. *Forwarder*\ s
+using *RESTful API*\ s can also be easily built as the *Drop Zone*
+naturally mappable to *REST*\ ful *API*\ s. The *Configuration
+Fragments* are the *REST* resources identified by the paths of the
+actual files. As the *Configuration Fragements* are stored in *JSON* or
+*XML* the *CRUD* operations are trivial.
+
+The *Installer*
+~~~~~~~~~~~~~~~
+
+The task of the\ *Installer* is to generate the actual configuration
+files, typically under /etc, for\ **\ the various *Ostro* software
+components. The *Installer* tracks configuration data changes in the
+*Drop Zone*\ using\ *inotify.*\ When a\ *Configuration
+Fragment*\ changed the *Installer* reads the altered configuration
+values and generates new versions of the effected configuration files
+incorporating the new values. The configuration file generation is
+template based.
+
+A single value change in a *Configuration Fragement* might lead to the
+re-generation of multiple configuration files of various *Ostro*
+components. For instance, if the listening port of the *SSH Daemon* is
+changed, the new value should be incorporated into the
+*/lib/systemd/system/sshd.socket*, */etc/sshd\_config* files as well as
```

Comment:
It's wording that "new value should be incorporated into the */lib/systemd/system/sshd.socket*" [file] which threw me off track. Make it clear here that this will get done by providing an override in /etc instead directly manipulating the file and the section is fine.

This gets explained later one, but without that knowledge, this section here can be misunderstood.